lib/_http_client.js

```diff
@@ -705,7 +705,7 @@ function onSocketNT(req, socket) {
 ClientRequest.prototype._deferToConnect = _deferToConnect;
 function _deferToConnect(method, arguments_, cb) {
   // This function is for calls that need to happen once the socket is
-  // connected and writable. It's an important promisy thing for all the socket
+  // assigned and writable. It's an important promisy thing for all the socket
```

Comment:
You might want to expand this a bit more. What does it mean for the socket to be assigned?

Reply:
`this.socket != null`?

Reply:
Yes, but why? Sockets are assigned to a request when they are available in the agent.

Reply:
expanded to follow the description from https://nodejs.org/api/http.html#http_event_socket